test/e2e/specs/specs-playwright/shared-specs/block-testing.ts

```diff
@@ -0,0 +1,88 @@
+import {
+	DataHelper,
+	BlockFlow,
+	setupHooks,
+	GutenbergEditorPage,
+	LoginFlow,
+	NewPostFlow,
+	EditorContext,
+	PublishedPostContext,
+} from '@automattic/calypso-e2e';
+import { Page } from 'playwright';
+
+/**
+ * Creates a suite of block smoke tests for a set of block flows.
+ *
+ * @param specName The parent name of the spec to use in the top-level describe. E.g. "CoBlocks"
+ * @param blockFlows A list of block flows to put under test.
+ */
+export function createBlockTests( specName: string, blockFlows: BlockFlow[] ): void {
+	describe( DataHelper.createSuiteTitle( specName ), function () {
+		let page: Page;
+		let gutenbergEditorPage: GutenbergEditorPage;
+		let editorContext: EditorContext;
+		let publishedPostContext: PublishedPostContext;
+
+		setupHooks( ( args ) => {
+			page = args.page;
+		} );
+
+		describe( 'Editor set up', function () {
+			it( 'Log in and start a new post', async function () {
+				const loginFlow = new LoginFlow( page, 'gutenbergSimpleSiteUser' );
+				await loginFlow.logIn();
+
+				const newPostFlow = new NewPostFlow( page );
+				await newPostFlow.newPostFromNavbar();
+
+				gutenbergEditorPage = new GutenbergEditorPage( page );
+				const title = DataHelper.getRandomPhrase();
+				await gutenbergEditorPage.enterTitle( title );
+				await gutenbergEditorPage.dismissWelcomeTourIfPresent();
```

Comment:
FYI in the viewport PR #56039 I made a change so that `dismissWelcomeTourIfPresent` method is called as part of the `newPostFlow.newPostFromNavbar` call.

Having it here shouldn't hurt anything, but it should no longer be necessary.